api/src/opentrons/protocol_api/labware.py

```diff
@@ -317,9 +324,25 @@ def uri(self) -> str:
 
     @property  # type: ignore[misc]
     @requires_version(2, 0)
-    def parent(self) -> LocationLabware:
+    def parent(self) -> Union[str, ModuleTypes, None]:
         """The parent of this labware. Usually a slot name."""
-        return self._implementation.get_geometry().parent.labware.object
+        if isinstance(self._implementation, LegacyLabwareCore):
+            # Type ignoring to preserve backwards compatibility
+            return self._implementation.get_geometry().parent.labware.object  # type: ignore
+
+        if self._protocol_core is None or self._core_map is None:
+            raise APIVersionError(
+                "Labware parent is not supported in this api version."
+            )
```

Comment:
Let's make this an `assert`, getting into this `if` would certainly be a bug on our part, not something the user could change.

```suggestion
        assert self._protocol_core and self._core_map, "Labware initialized incorrectly"
```